### PR TITLE
Introduce drag and drop support

### LINKED ISSUE #1744: Add missing tests drag-and-drop.spec.ts
drag-and-drop.spec.ts: Input.drag: should throw an exception if not enabled before usage
drag-and-drop.spec.ts: Input.drag: should emit a dragIntercepted event when dragged
drag-and-drop.spec.ts: Input.drag: should emit a dragEnter
drag-and-drop.spec.ts: Input.drag: should emit a dragOver event
drag-and-drop.spec.ts: Input.drag: can be dropped
drag-and-drop.spec.ts: Input.drag: can be dragged and dropped with a single function
drag-and-drop.spec.ts: Input.drag: can be disabled

### PR DESCRIPTION
**Pull request by kblok**
closes #1744

### CODE REVIEW

**Comment by kgar** [comment]
I'm glad to review this. It'll take me a little time to read and understand.

File: lib/PuppeteerSharp.Tests/DragAndDropTests/InputDragTests.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        [Fact(Timeout = TestConstants.DefaultTestTimeout)]
+        public async Task CanBeDraggedAndDroppedWithASingleFunction()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/drag-and-drop.html");
+            Assert.False(Page.IsDragInterceptionEnabled);
+            await Page.SetDragInterceptionAsync(true);
+            Assert.True(Page.IsDragInterceptionEnabled);
+            var draggable = await Page.QuerySelectorAsync("#drag");
+            var dropzone = await Page.QuerySelectorAsync("#drop");
+            await draggable.DragAndDropAsync(dropzone);
```
Comment (kgar) [inline lib/PuppeteerSharp.Tests/DragAndDropTests/InputDragTests.cs:109]: 🔥 this is the stuff

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -2529,9 +2538,8 @@ public void Dispose()
         /// the garbage collector can reclaim the memory that the <see cref="Page"/> was occupying.</remarks>
         /// <param name="disposing">Indicates whether disposal was initiated by <see cref="Dispose()"/> operation.</param>
         protected virtual void Dispose(bool disposing) => _ = DisposeAsync();
-        #endregion
```
Comment (kgar) [inline lib/PuppeteerSharp/Page.cs:2541]: I recommend removing a newline to satisfy the Style Cop rule for newlines.

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: done!

File: lib/PuppeteerSharp/Input/Mouse.cs [hunk 3]
```diff
@@ -149,5 +149,114 @@ public Task WheelAsync(decimal deltaX, decimal deltaY)
...
+        /// <summary>
+        /// Performs a drag, dragenter, dragover, and drop in sequence.
+        /// </summary>
+        /// <param name="startX">Start X coordinate</param>
+        /// <param name="startY">Start Y coordinate</param>
+        /// <param name="endX">End X coordinate</param>
+        /// <param name="endY">End Y coordinate</param>
+        /// <param name="delay">If specified, is the time to wait between `dragover` and `drop` in milliseconds.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
```
Comment (kgar) [inline lib/PuppeteerSharp/Input/Mouse.cs:247]: Is this `NotImplementedException` tag still needed in the xml doc?

Comment (kblok) [reply lib/PuppeteerSharp/Input/Mouse.cs]: No. Removed :)

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -369,6 +367,11 @@ public int DefaultTimeout
         /// </summary>
         public Accessibility Accessibility { get; }
 
+        /// <summary>
+        /// `true` if drag events are being intercepted, `false` otherwise.
+        /// </summary>
+        public bool IsDragInterceptionEnabled { get; internal set; }
```
Comment (kgar) [inline lib/PuppeteerSharp/Page.cs:373]: I recommend making the setter `private set;`

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: done!

**Review by kgar — CHANGES_REQUESTED** [review]
Disclaimer: This is my first ever review on Github. Thank you for letting me be a part of this.

I have a few suggestions in the comments.
Of those suggestions, I get an analyzer-based compilation error because of the additional newline on `Page.cs`.

**Comment by kblok** [comment]
>Disclaimer: This is my first ever review on Github. Thank you for letting me be a part of this.
>I have a few suggestions in the comments.
>Of those suggestions, I get an analyzer-based compilation error because of the additional newline on Page.cs.

You're welcome! It's always good to get some fresh eyes. I addressed all your comments.

Thanks

**Review by kgar — APPROVED** [review]
🎉🎉🎉

**Pull request merged**